### ISSUE TITLE
Epoch compile button does not recreate the AST

### ISSUE THREAD

**Issue by edwardalee** [lf-lang/lingua-franca#746]
It turns out that with the new Epoch button to compile the program, the button does not cause a new AST to get created.  So code gets generated on the basis of the previous AST that was present, including any AST transformations that were previously applied.

This is a serious problem for federated execution, but could be a problem for non-federated programs using `after`. One way to see the effect is to compile a program that has an `after` in it. Then close the diagram view and re-open the diagram view using the `Window->Show View` menu. You will see the transformed program instead of the original one.

For federated programs, this results in code being generated that has no communication between federates.

Not sure how to solve this problem...

**Comment by edwardalee** [comment]
Related issue: In Epoch, we have (thanks to @a-sr ) a button to generate code for a program.  In a federated program, such as `test/c/src/federated/HelloFederated.lf`, if you push this button twice without modifying the code, the first generated program works, but the second one segfaults.

This happens at least in the scalability-banks branch, but I suspect it happens in all branches.

The culprit is this line in GeneratorBase:
```
            // Remove the connections at the top level
            mainReactor.connections.clear()
```
The problem is that this line modifies the E-core model, and the second button push uses the modified E-core model to generate code.

A simple fix would be to augment the button added by @a-sr to first regenerate the E-core model, as happens when you make any modification to the file.  Any idea how to do that?

**Comment by a-sr** [comment]
I was able to identify my mistake. If compiling the content of the active editor, I retrieved the resource/model from the editor directly. Hence, the code generator, editor, and diagram synthesis were working in the same model. But I forgot about the transformation during compilation that then leaked into the diagram and future compilations. With #848 each compilation run will get its own separate model.
